cmd/lifecycle/analyzer.go

```diff
@@ -208,6 +214,51 @@ func (aa analyzeArgs) analyze() (platform.AnalyzedMetadata, error) {
 	return analyzedMD, nil
 }
 
+func (a *analyzeCmd) validateStack() error {
+	if !a.supportsStackValidation() {
+		return nil
+	}
+
+	var stackMD platform.StackMetadata
+	if _, err := toml.DecodeFile(a.stackPath, &stackMD); err != nil && !os.IsNotExist(err) {
```

Comment:
What should happen if the file doesn't exist? Don't we want to handle it or at least add a comment about it?

Reply:
I think we fall back to `CNB_STACK_ID`.

Reply:
What about making a `lifecycle.ResolveStackID` function?

Reply:
Regarding what happens if the file doesn't exist:
I think that if the file doesn't exist, `stackMD` will be empty and then in `getRunImage`, if `a.runImageRef == ""`, `ResolveRunImage` will return an error.
Maybe it's worth adding a comment about it?

Reply:
If the file doesn't exist, it's still possible that `a.runImageRef` is non-empty (specified via command flag or env var).

Reply:
Yeah, I know that it's possible but if `a.runImageRef` is also empty, then we might get an error.
Anyway, I think that maybe it's worth adding a comment about it.

Reply:
I'm not sure it's necessary, but I wouldn't be opposed to adding a comment. @jabrown85 WDYT?

Reply:
I'm open to a suggestion with the comment we think is appropriate. I am far less comment driven than test driven - I find comments fall out of date rather quickly unless it's describing a public API.

Reply:
Let's leave it for now, we could add a comment if we find that it's necessary.